pkg/account/api/account.go

```diff
@@ -128,6 +135,155 @@ func (s *AccountService) CreateAccountV2(
 	return &accountproto.CreateAccountV2Response{Account: account.AccountV2}, nil
 }
 
+func (s *AccountService) createAccountV2NoCommand(
+	ctx context.Context,
+	req *accountproto.CreateAccountV2Request,
+	localizer locale.Localizer,
+	editor *eventproto.Editor,
+) (*accountproto.CreateAccountV2Response, error) {
+	err := validateCreateAccountV2NoCommandRequest(req, localizer)
+	if err != nil {
+		s.logger.Error(
+			"Failed to create account",
+			log.FieldsFromImcomingContext(ctx).AddFields(
+				zap.Error(err),
+				zap.String("organizationID", req.OrganizationId),
+			)...,
+		)
+		return nil, err
+	}
+	account := domain.NewAccountV2(
+		req.Email,
+		req.Name,
+		req.FirstName,
+		req.LastName,
+		req.Language,
+		req.AvatarImageUrl,
+		req.OrganizationId,
+		req.OrganizationRole,
+		req.EnvironmentRoles,
+	)
+	err = s.accountStorage.RunInTransaction(ctx, func() error {
+		// TODO: temporary implementation: double write account v2 ---
+		exist, err := s.accountStorage.GetAccountV2(ctx, account.Email, req.OrganizationId)
+		if err != nil && !errors.Is(err, v2as.ErrAccountNotFound) {
+			return err
+		}
+		if exist != nil {
+			return s.changeExistedAccountV2EnvironmentRoles(ctx, req, exist, editor)
+		}
+		// TODO: temporary implementation end ---
+
+		createAccountEvent, err := domainevent.NewEvent(
+			editor,
+			eventproto.Event_ACCOUNT,
+			account.Email,
+			eventproto.Event_ACCOUNT_V2_CREATED,
+			&eventproto.AccountV2CreatedEvent{
+				Email:            account.Email,
+				FirstName:        account.FirstName,
+				LastName:         account.LastName,
+				Language:         account.Language,
+				AvatarImageUrl:   account.AvatarImageUrl,
+				OrganizationId:   account.OrganizationId,
+				OrganizationRole: account.OrganizationRole,
+				EnvironmentRoles: account.EnvironmentRoles,
+				Disabled:         account.Disabled,
+				CreatedAt:        account.CreatedAt,
+				UpdatedAt:        account.UpdatedAt,
+			},
+			storage.AdminEnvironmentNamespace,
+			account,
+			nil,
+		)
+		if err != nil {
+			return err
+		}
+		if err = s.publisher.Publish(ctx, createAccountEvent); err != nil {
+			return err
+		}
```

Comment:
@hvn2k1 
I think it would be better to send this process when the DB operation is successfully completed, rather than during the transaction.

What do you think?

Reply:
Actually, I don't know what should be prioritized in this situation :))) because if the sending to audit log success and db operation failed, at least we know that someone was trying to do the action, but if the sending to audit log failed first then the db operation is not commit.
On the reverse, the DB operation is committed first but if it was a very important log like delete the account, should that be ok?
Can you help me with this too? cc @cre8ivejp @kentakozuka @Ubisoft-potato

Reply:
currently other APIs are pushing the event first then commit DB, maybe we are prioritizing audit log!

Reply:
@hvn2k1 Yes, the current implementation is sending auditlog during the transaction, so we can konw there is someone want to do the operation, no matter the operation is successed.

Reply:
@hvn2k1, cc: @kakcy
Unfortunately, to improve this requires a lot of refactoring, so for now, let's do like the feature API is doing to keep the consistency. It publishes the event after the transaction.

Reference:
https://github.com/bucketeer-io/bucketeer/blob/main/pkg/feature/api/feature.go#L710
https://github.com/bucketeer-io/bucketeer/blob/main/pkg/feature/api/feature.go#L792-L809
https://github.com/bucketeer-io/bucketeer/blob/main/pkg/feature/api/feature.go#L826-L842

Reply:
I have switched publish the event after the transaction!